kernel/dw1000.c

```diff
@@ -1809,10 +1821,10 @@ static int dw1000_rx_prepare(struct dw1000 *dw)
 			 &rx->time, sizeof(rx->time));
 	dw1000_init_read(&rx->info, &rx->rx_fqual, DW1000_RX_FQUAL, 0,
 			 &rx->fqual, sizeof(rx->fqual));
-	dw1000_init_read(&rx->info, &rx->sys_status, DW1000_SYS_STATUS,
-			 DW1000_SYS_STATUS2, &rx->rxovrr, sizeof(rx->rxovrr));
 	dw1000_init_read(&rx->info, &rx->dig_diag, DW1000_DIG_DIAG,
 			 DW1000_EVC_OVR, &rx->evc_ovr, sizeof(rx->evc_ovr));
+	dw1000_init_read(&rx->info, &rx->sys_status, DW1000_SYS_STATUS, 0,
+			 &rx->status, sizeof(rx->status));
```

Comment:
What's the reason for switching the order in which we read SYS_STATUS and DIG_DIAG?